### DISCUSSION TITLE
View bed with eeprom data

### DISCUSSION THREAD

**Discussion by RubiXKube** [jneilliii/OctoPrint-BedLevelVisualizer#419] [Ideas]
When using my printer bed move and I'm using G29 on each print
Mesh is always save, so when i want to use bed visualizer, i have to do G29 again
to know how is my actual bed, but there is a really faster way to do it

I swap my entire gcode to "M420 V" and then click on "Update Mesh Now" !

Is it possible to have a second button with "Eeprom Data" or anything else like that ?

**Comment by jneilliii** [comment]
Check in the plugin's settings, there is a tab for creating custom buttons...you can put anything in there and if you want to have the plugin visualize the mesh, make sure to add the `@BEDLEVELVISUALIZER` token before the `M420 V`. I actually use the custom buttons to do the different steps of UBL leveling, ie G29 P1, G29 P3, G29 S, etc. and leave the main update mesh button to M420 V like you mention.

**Comment by jneilliii** [comment]
if you G29 on each print, you could also add either `M118 E1 BEDLEVELVISUALIZER` or `@BEDLEVELVISUALIZER` to your start gcode just prior to the reporting of the bed mesh and it will auto update the stored mesh in the plugin.

**Reply by jes00** [reply]
Just change 
> G29
 
to
> @BEDLEVELVISUALIZER G29

Right?

**Reply by jneilliii** [reply]
Yep

**Comment by RubiXKube** [comment]
Perfect, thank you
I've tried the custom button but i didin't understood how to use @BEDLELVISUALIZER command

**Comment by genepaton** [comment]
Hey guys,

I've forgotten a lot of my self taught understanding of Octoprint, customised firmware and Marlin commands. 
I'm running a custom Marlin (bilinear enabled i'm pretty positive) firmware based of Jyers on my Ender 3 V2.
The automatic leveling using my CRTouch and mesh creation is terrible, so once I've fine tuned it manually and saved it, I never adjust it again. However I think I also have an OCD issue, and would really like to see a mapping of my currently saved mesh so I can waste hours trying to get it almost perfectly level.
This plugin is great, however can I replace the line that starts the Mesh creation 'G29 T' with something to just load the mesh in memory and then create the visualisation off of that?

**Reply by jneilliii** [reply]
Yes, you can use `M420 V` to load the current mesh from memory.
